src/server/chunk_server/chunk_server_impl.cc

```diff
@@ -0,0 +1,129 @@
+#include "src/server/chunk_server/chunk_server_impl.h"
+
+#include "absl/strings/str_cat.h"
+#include "absl/time/clock.h"
+#include "src/common/system_logger.h"
+
+using gfs::common::ConfigManager;
+using gfs::common::thread_safe_flat_hash_map;
+using gfs::service::ChunkServerServiceChunkServerClient;
+using gfs::service::MasterChunkServerManagerServiceClient;
+using google::protobuf::util::Status;
+using google::protobuf::util::StatusOr;
+
+namespace gfs {
+namespace server {
+
+ChunkServerImpl::~ChunkServerImpl() {
+  master_server_clients_.clear();
```

Comment:
I think the default destructor should call the destructor of these collections and thus clean them up?